### ISSUE TITLE
repo.addDomain reseting other domain state

### ISSUE THREAD

**Issue by greypants** [vigetlabs/microcosm#259]
**This issue popped up when upgrading from `11.6.0` to `12.2.0`**

Domain state is getting reset to the result of `getInitialState` for some reason after calling `addDomain`:

```js
export default class LicenseManager extends Presenter {
  setup(repo, props) {
    console.log('LicenseManager BEFORE addDomain', repo.state.user)
    repo.addDomain('licenses', LicensesDomain)
    console.log('LicenseManager AFTER addDomain', repo.state.user)
  }
  …
```
Logs the following:

![logs](https://cloud.githubusercontent.com/assets/647549/23967476/710daa7c-0995-11e7-9b6c-6774d08478ad.png)

The second log is the initial state of the `user` domain.

This `Presenter` is nested under two other parent presenters.

**Comment by nhunzaker** [comment]
This should be fixed in 12.2.1, currently in release candidate. Will close after we publish to stable.

**Comment by nhunzaker** [comment]
Fixed https://github.com/vigetlabs/microcosm/releases/tag/v12.2.1
